File: app/src/main/java/com/sendbird/calls/quickstart/SplashActivity.java
package com.sendbird.calls.quickstart;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.sendbird.calls.quickstart.utils.ActivityUtils;
import com.sendbird.calls.quickstart.utils.AuthenticationUtils;
import com.sendbird.calls.quickstart.utils.ToastUtils;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends AppCompatActivity {

    private static final int SPLASH_TIME_MS = 1000;

    private Context mContext;
    private Timer mTimer;
    private Boolean mAutoAuthenticateResult;
    private String mEncodedAuthInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        mContext = this;

        setTimer();

        if (!hasDeepLink()) {
            autoAuthenticate();
        }
    }

    private boolean hasDeepLink() {
        boolean result = false;

        Intent intent = getIntent();
        if (intent != null) {
            Uri data = intent.getData();
            if (data != null) {
                String scheme = data.getScheme();
                if (scheme != null && scheme.equals("sendbird")) {
                    Log.i(BaseApplication.TAG, "[SplashActivity] deep link: " + data.toString());
                    mEncodedAuthInfo = data.getHost();
                    if (!TextUtils.isEmpty(mEncodedAuthInfo)) {
                        result = true;
                    }
                }
            }
        }
        return result;
    }

    private void setTimer() {
        mTimer = new Timer();
        mTimer.schedule(new TimerTask() {
            @Override
            public void run() {
                runOnUiThread(() -> {
                    mTimer = null;

                    if (!TextUtils.isEmpty(mEncodedAuthInfo)) {
                        AuthenticationUtils.authenticateWithEncodedAuthInfo(SplashActivity.this, mEncodedAuthInfo, (isSuccess, hasInvalidValue) -> {
                            if (isSuccess) {
                                ActivityUtils.startMainActivityAndFinish(SplashActivity.this);
                            } else {
                                if (hasInvalidValue) {
                                    ToastUtils.showToast(SplashActivity.this, getString(R.string.calls_invalid_deep_link));
                                } else {
                                    ToastUtils.showToast(SplashActivity.this, getString(R.string.calls_deep_linking_to_authenticate_failed));
                                }
                                finish();
                            }
                        });
                        return;
                    }

                    if (mAutoAuthenticateResult != null) {
                        if (mAutoAuthenticateResult) {
                            ActivityUtils.startMainActivityAndFinish(SplashActivity.this);
                        } else {
                            ActivityUtils.startAuthenticateActivityAndFinish(SplashActivity.this);
                        }
                    }
                });
            }
        }, SPLASH_TIME_MS);
    }

    private void autoAuthenticate() {
        AuthenticationUtils.autoAuthenticate(mContext, userId -> {
            if (mTimer != null) {
                mAutoAuthenticateResult = !TextUtils.isEmpty(userId);
            } else {
                if (userId != null) {
                    ActivityUtils.startMainActivityAndFinish(SplashActivity.this);
                } else {
                    ActivityUtils.startAuthenticateActivityAndFinish(SplashActivity.this);
                }
            }
        });
    }

    @Override
    public void onBackPressed() {
        if (mTimer != null) {
            mTimer.cancel();
            mTimer = null;
        }
        super.onBackPressed();
    }
}

File: directcall-kotlin/app/src/main/java/com/sendbird/calls/quickstart/SplashActivity.kt
package com.sendbird.calls.quickstart

import android.os.Bundle
import android.util.Log
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.AppCompatActivity
import com.sendbird.calls.quickstart.utils.AuthenticationUtils
import com.sendbird.calls.quickstart.utils.showToast
import com.sendbird.calls.quickstart.utils.startAuthenticateActivityAndFinish
import com.sendbird.calls.quickstart.utils.startMainActivityAndFinish
import java.util.*

const val SPLASH_TIME_MS = 1000L

class SplashActivity : AppCompatActivity() {
    private var mTimer: Timer? = null
    private var mAutoAuthenticateResult: Boolean = false
    private var mEncodedAuthInfo: String? = null
    private val backPressedCallback = object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
            if (mTimer != null) {
                mTimer?.cancel()
                mTimer = null
            }
            finish()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        this.onBackPressedDispatcher.addCallback(backPressedCallback)
        setTimer()
        if (!hasDeepLink()) {
            autoAuthenticate()
        }
    }

    private fun hasDeepLink(): Boolean {
        val data = intent.data ?: return false
        val scheme = data.scheme
        if (scheme != null && scheme == "sendbird") {
            Log.i(TAG, "[SplashActivity] deep link: $data")
            mEncodedAuthInfo = data.host
            return !(mEncodedAuthInfo.isNullOrEmpty())
        }
        return false
    }

    private fun setTimer() {
        mTimer = Timer().apply {
            this.schedule(object : TimerTask() {
                override fun run() {
                    runOnUiThread {
                        mTimer = null
                        if (!mEncodedAuthInfo.isNullOrEmpty()) {
                            AuthenticationUtils.authenticateWithEncodedAuthInfo(this@SplashActivity, mEncodedAuthInfo) { isSuccess, hasInvalidValue ->
                                if (isSuccess) {
                                    startMainActivityAndFinish()
                                } else {
                                    if (hasInvalidValue) {
                                        showToast(getString(R.string.calls_invalid_deep_link))
                                    } else {
                                        showToast(getString(R.string.calls_deep_linking_to_authenticate_failed))
                                    }
                                    finish()
                                }
                            }
                            return@runOnUiThread
                        }
                        if (mAutoAuthenticateResult) {
                            startMainActivityAndFinish()
                        } else {
                            startAuthenticateActivityAndFinish()
                        }
                    }
                }
            }, SPLASH_TIME_MS)
        }
    }

    private fun autoAuthenticate() {
        AuthenticationUtils.autoAuthenticate(this) { userId ->
            if (mTimer != null) {
                mAutoAuthenticateResult = !userId.isNullOrEmpty()
            } else {
                if (userId != null) {
                    startMainActivityAndFinish()
                } else {
                    startAuthenticateActivityAndFinish()
                }
            }
        }
    }
}
